runtime/vm/resolvesupport.cpp

```diff
@@ -973,11 +973,10 @@ resolveInstanceFieldRefInto(J9VMThread *vmStruct, J9Method *method, J9ConstantPo
 		 * since ValueType superclasses can not have fields.
 		 */
 #if defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)
-		if (J9_IS_J9CLASS_VALUETYPE(resolvedClass)) {
-			if ('Q' == J9UTF8_DATA(signature)[0]) {
-				flattenedClassCache = resolvedClass->flattenedClassCache;
-				fieldIndex = findIndexInFlattenedClassCache(flattenedClassCache, nameAndSig);
-				Assert_VM_false(UDATA_MAX == fieldIndex);
+		if (NULL != resolvedClass->flattenedClassCache) {
+			flattenedClassCache = resolvedClass->flattenedClassCache;
+			fieldIndex = findIndexInFlattenedClassCache(flattenedClassCache, nameAndSig);
+			if (UDATA_MAX != fieldIndex) {
```

Comment:
`definingClass = resolvedClass;` may not be correct if its not a VT

Reply:
Right, added the VT check back in.